internal/render/login/statefulset.go

```diff
@@ -101,6 +101,7 @@ func RenderStatefulSet(
 					DNSConfig: &corev1.PodDNSConfig{
 						Searches: []string{
 							naming.BuildServiceFQDN(consts.ComponentTypeWorker, namespace, clusterName),
+							naming.BuildLoginHeadlessServiceFQDN(namespace, clusterName),
```

Comment:
You need to add same (`soperator-login-headless-svc.soperator.svc.cluster.local`) to the worker statefulset